libs/ml/src/utilities/sparse_tensor_utilities.cpp

```diff
@@ -0,0 +1,289 @@
+
+    // Parallel addition
+    dst_slice.in_parallel().RangedApplyMultiple(
+        range, [](auto const &a, auto const &b, auto &c) { c = b + a; }, dst_slice, src_slice);
+    dst_index++;
+  }
+
+  return std::move(dst);
+}
+
+template <class TensorType>
+TensorType FromSparse(TensorType const &                               src,
+                      std::unordered_set<fetch::math::SizeType> const &update_rows,
+                      fetch::math::SizeType                            output_rows)
+{
+  using SizeType = fetch::math::SizeType;
+
+  TensorType dst({src.shape().at(0), output_rows});
+
+  SizeType src_index = 0;
+  for (SizeType dst_index : update_rows)
+  {
+    auto dst_view = dst.View(dst_index);
+    auto src_view = src.View(src_index);
+    dst_view.Assign(src_view);
+
+    src_index++;
+  }
+
+  return std::move(dst);
+}
+
+template void SparseAdd<math::Tensor<int8_t>>(
+    math::Tensor<int8_t> const &src, math::Tensor<int8_t> &dst,
+    std::unordered_set<fetch::math::SizeType> const &update_rows,
+    fetch::math::SizeType                            sparsity_threshold);
+template void SparseAdd<math::Tensor<int16_t>>(
+    math::Tensor<int16_t> const &src, math::Tensor<int16_t> &dst,
+    std::unordered_set<fetch::math::SizeType> const &update_rows,
+    fetch::math::SizeType                            sparsity_threshold);
+template void SparseAdd<math::Tensor<int32_t>>(
+    math::Tensor<int32_t> const &src, math::Tensor<int32_t> &dst,
+    std::unordered_set<fetch::math::SizeType> const &update_rows,
+    fetch::math::SizeType                            sparsity_threshold);
+template void SparseAdd<math::Tensor<int64_t>>(
+    math::Tensor<int64_t> const &src, math::Tensor<int64_t> &dst,
+    std::unordered_set<fetch::math::SizeType> const &update_rows,
+    fetch::math::SizeType                            sparsity_threshold);
+template void SparseAdd<math::Tensor<uint8_t>>(
+    math::Tensor<uint8_t> const &src, math::Tensor<uint8_t> &dst,
+    std::unordered_set<fetch::math::SizeType> const &update_rows,
+    fetch::math::SizeType                            sparsity_threshold);
+template void SparseAdd<math::Tensor<uint16_t>>(
+    math::Tensor<uint16_t> const &src, math::Tensor<uint16_t> &dst,
+    std::unordered_set<fetch::math::SizeType> const &update_rows,
+    fetch::math::SizeType                            sparsity_threshold);
+template void SparseAdd<math::Tensor<uint32_t>>(
+    math::Tensor<uint32_t> const &src, math::Tensor<uint32_t> &dst,
+    std::unordered_set<fetch::math::SizeType> const &update_rows,
+    fetch::math::SizeType                            sparsity_threshold);
+template void SparseAdd<math::Tensor<uint64_t>>(
+    math::Tensor<uint64_t> const &src, math::Tensor<uint64_t> &dst,
+    std::unordered_set<fetch::math::SizeType> const &update_rows,
+    fetch::math::SizeType                            sparsity_threshold);
```

Comment:
based on @markos 's comments i think maybe we shouldn't add the uint cases; please comment these out for now

Reply:
but uint instantiations needed for ml/ops/variable.cpp, because it also has uint instantiations. So if I comment them in utils I have to comment them in variable.cpp.

Reply:
even everywhere in ops where uint instantiated